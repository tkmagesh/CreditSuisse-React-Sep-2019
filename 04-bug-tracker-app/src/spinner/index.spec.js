import { spinnerReducer } from './index';
import { Spinner } from './index';
import { shallow } from 'enzyme';
import React from 'react';

describe('Spinner Reducer', () => {
	it('increments the current state by 1 for UP action', () => {
		//Arrange
			let action = { type : 'UP' };
			let expectedResult = 1;
				
		//Act
			let result = spinnerReducer(0, action)

		//Assert
			expect(result).toBe(expectedResult);
	});
});

describe('Spinner snapshot testing', () => {
	it('should match the snapshot', () => {

		  const component = shallow(<Spinner spinnerValue={0} up={()=>{}} down={()=>{}} doubleUp={()=>{}} doubleDown={()=>{}}/>);
		  expect(component).toMatchSnapshot();

	})

	it('should call the up action', () => {
		const upActionMock = jest.fn();

		const component = shallow(<Spinner spinnerValue={0} up={upActionMock} down={()=>{}} doubleUp={()=>{}} doubleDown={()=>{}}/>);

		component.find('input[type="button"][value="Increment"]').simulate('click');
		expect(upActionMock).toHaveBeenCalled();
	})
})