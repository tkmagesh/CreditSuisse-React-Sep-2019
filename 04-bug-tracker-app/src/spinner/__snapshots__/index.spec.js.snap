// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner snapshot testing should match the snapshot 1`] = `
<div>
  <span>
    asdfdsafas
  </span>
  <input
    onClick={[Function]}
    type="button"
    value="Double Decrement"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Decrement"
  />
  <span>
     [ 
    0
     ] 
  </span>
  <input
    onClick={[Function]}
    type="button"
    value="Increment"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Double Increment"
  />
</div>
`;
